src/components/Notice/Notice.jsx

```diff
@@ -0,0 +1,78 @@
+/* eslint react/no-array-index-key: 0 */
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import { Alert, Button, Container } from 'react-bootstrap'
+
+const Notice = ({ alert, buttonProps, dismissible, withBackButton }) => {
+  const [show, setShow] = useState(true)
+  const { title, body, variant } = alert
+  let onClick, text
+  if (withBackButton) ({ onClick, text } = buttonProps)
+
+  return (
+    show && (
+      <Container>
+        <Alert className='my-5 text-break' variant={variant} onClose={() => setShow(false)} dismissible={dismissible}>
+          {title && (
+            <Alert.Heading>{title}</Alert.Heading>
+          )}
+          {body.map((text, index) => {
+            // If "text" is a JSON string, return it as a formatted code block
+            // Otherwise, return it as a regular string
+            const isJSONParsable = () => {
+              try {
+                JSON.parse(text)
+                return true
+              } catch (error) {
+                return false
+              }
+            }
+            const isJavascriptObject = isJSONParsable(text)
+
+            return (
+              <>
+                {isJavascriptObject ? (
+                  <pre className='mb-0' key={index}>{text}</pre>
+                ) : (
+                  <p className='mb-0' key={index}>{text}</p>
+                )}
+              </>
+            )
```

Comment:
@summer-cook I added this part after you approved the pr. can you take another look?

Reply:
@alishaevn looks good!